src/app/space/app-launcher/services/app-launcher-work-space.service.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { TestBed } from '@angular/core/testing';
+import { Observable, of } from 'rxjs';
+import { WorkspaceLinks } from './../../create/codebases/services/workspace';
+import { AppLaunchWorkSpaceService } from './app-launcher-work-space.service';
+
+describe('AppLaunchWorkSpaceService: ', () => {
+  const mockCreateWorkSpace: WorkspaceLinks = {
+    links: {
+      open: 'https://che.openshift.io/viraj-1/sep-3-app-irtde'
+    }
+  };
+
+  const mockAppLaunchWorkSpaceService = {
+    createWorkSpace(): Observable<WorkspaceLinks> {
+      return of(mockCreateWorkSpace);
+    }
+  };
+
+  let appLaunchWorkSpaceService: AppLaunchWorkSpaceService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {provide: AppLaunchWorkSpaceService, useValue: mockAppLaunchWorkSpaceService}
+      ]
+    });
+    appLaunchWorkSpaceService = TestBed.get(AppLaunchWorkSpaceService);
+  });
+
+  it('Create workSpace', async () => {
+    appLaunchWorkSpaceService.createWorkSpace('').subscribe((val: WorkspaceLinks) => {
+      expect(val).toEqual(mockCreateWorkSpace);
```

Comment:
What is the meaning of `''`  (empty string parameter) here? Can we structure the tests in a form of functionality (and boundary cases) which this service should expose?

Reply:
Here is the unit test for ```workSpaceService```https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/codebases/services/workspaces.service.spec.ts.

> Can we structure the tests in a form of functionality (and boundary cases) which this service should expose?

Yes, we can write the tests in the form of functionality. But, those tests should go in ```ngx-launcher``` because the component that uses this service is in ```ngx-launcher```.

Reply:
That's fine - can you contribute those tests there and remove this one?